apps/api/v2/src/ee/event-types/event-types_2024_06_14/controllers/event-types.controller.ts

```diff
@@ -117,26 +140,26 @@ export class EventTypesController_2024_06_14 {
     if (!eventTypes || eventTypes.length === 0) {
       throw new NotFoundException(`Event types not found`);
     }
-    const eventTypesFormatted = this.eventTypeResponseTransformPipe.transform(eventTypes);
-    const eventTypesWithoutHiddenFields = eventTypesFormatted.map((eventType) => {
+    const eventTypesWithoutHiddenEventTypes = eventTypes.filter((eventType) => !eventType.hidden);
+    const eventTypesFormatted = this.eventTypeResponseTransformPipe.transform(
+      eventTypesWithoutHiddenEventTypes
+    );
+    const eventTypesWithoutHiddenFieldsAndHiddenEventTypes = eventTypesFormatted.map((eventType) => {
```

Comment:
@ThyMinimalDev - since atoms hooks send authentication with the request (i think) we should preserve the current logic - if auth is provided return all hidden and non hidden event types belonging to that user. If auth of another user is provided (user X) who is fetching event types for user Y then don't return the event types that are hidden.

Reply:
I also don't see the point in filtering out hidden booking fields - the problem is that we return hidden event types and we only need to filter out event types.